atomic_reactor/plugins/pre_yum_proxy.py

```diff
@@ -0,0 +1,162 @@
+                        proxied_baseurl = urlunsplit(
+                            (parts[0], self.yum_proxies[parts.netloc]) + parts[2:])
+                        yum_repo_file_content.set(yum_repo_name, 'baseurl', proxied_baseurl)
+                        rewrite_file = True
+
+                if rewrite_file:
+                    dest_path = os.path.join(self.workdir, 'updated', parent_image_id,
+                                             yum_repo_file_name)
+                    self._write_yum_repo_file(dest_path, yum_repo_file_content)
+                    self.updated_yum_repo_files.setdefault(parent_image_id, [])
+                    self.updated_yum_repo_files[parent_image_id].append(yum_repo_file_name)
+
+    def _write_yum_repo_file(self, dest_path, content):
+        try:
+            os.makedirs(os.path.dirname(dest_path))
+        except OSError:
+            # Ignore if dirs have already been created. Python 2 doesn't suport exists_ok param
+            pass
+        with open(dest_path, 'w') as f:
+            for yum_repo_name in content.sections():
+                f.write('[{}]\n'.format(yum_repo_name))
+                for key, value in content.items(yum_repo_name):
+                    f.write('{} = {}\n'.format(key, value))
+
+    def _update_dockerfile(self):
+        dfp = df_parser(self.workflow.builder.df_path)
+
+        for parent_image_id, yum_repo_files in self.updated_yum_repo_files.items():
+            if not yum_repo_files:
+                continue
+
+            parent_structure = self._find_parent_structure(dfp, parent_image_id)
+            update_lines = []
+            for yum_repo_file_name in yum_repo_files:
+                update_path = os.path.join(self.workdir, 'updated', parent_image_id,
+                                           yum_repo_file_name)
+                update_lines.append('ADD {} /etc/yum.repos.d/'.format(update_path))
+            dfp.add_lines_at(parent_structure, *update_lines, after=True)
+
+            last_stage_structure = self._find_last_stage_structure(dfp, parent_structure)
+            original_lines = []
+            for yum_repo_file_name in yum_repo_files:
+                original_path = os.path.join(self.workdir, 'original', parent_image_id,
+                                             yum_repo_file_name)
+                original_lines.append('ADD {} /etc/yum.repos.d/'.format(original_path))
+            dfp.add_lines_at(last_stage_structure, *original_lines, after=True)
+
+    def _find_parent_structure(self, dfp, parent_image_id):
+        for structure in dfp.structure:
+            if structure['instruction'] != 'FROM':
+                continue
+            found_parent_image_id = structure['value'].split(' ')[0]
+            if found_parent_image_id == parent_image_id:
+                return structure
+
+        raise RuntimeError('Unable to find parent image instruction')
+
+    def _find_last_stage_structure(self, dfp, parent_structure):
```

Comment:
Could use a docstring. I original thought this method was intending to find the start of the final stage, but it seems to actually be looking for the end of the current stage.